lib/ansible/modules/net_tools/ip_netns.py

```diff
@@ -22,95 +22,175 @@
 short_description: Manage network namespaces
 requirements: [ ip ]
 description:
     - Create or delete network namespaces using the ip command.
 options:
     name:
-        required: false
+        required: True
         description:
             - Name of the namespace
+    interfaces:
+        version_added: 2.6
+        required: False
+        default: []
+        description:
+            - List of interface names that need to be part of the namespace
+    operation:
+        version_added: 2.6
+        required: False
+        choices: ['add', 'remove']
+        default: add
+        description:
+            - Whether interfaces should be add/deleted to/from namespace
     state:
-        required: false
-        default: "present"
+        required: True
         choices: [ present, absent ]
         description:
             - Whether the namespace should exist
-'''
+"""
 
-EXAMPLES = '''
+EXAMPLES = """
 # Create a namespace named mario
 - name: Create a namespace named mario
-  namespace:
+  ip_netns:
+    name: mario
+    state: present
+# Create a namespace name mario and add eth1, eth2 interfaces to it
+- name: Create a namespace named mario and add interfaces
+  ip_netns:
     name: mario
+    interfaces:
+        - eth1
+        - eth2
+    operation: add
     state: present
 - name: Delete a namespace named luigi
-  namespace:
+  ip_netns:
     name: luigi
     state: absent
-'''
+- name: Create a namespace named mario and remove interfaces
+  ip_netns:
+    name: mario
+    interfaces:
+        - eth1
+        - eth2
+    operation: remove
+    state: present
+"""
 
-RETURN = '''
+RETURN = """
 # Default return values
-'''
+"""
 
+import re
 from ansible.module_utils.basic import AnsibleModule
 
 
 class Namespace(object):
-    """Interface to network namespaces. """
+    """ Interface to network namespaces. """
 
     def __init__(self, module):
         self.module = module
         self.name = module.params['name']
+        self.interfaces = module.params['interfaces']
+        self.operation = module.params['operation']
         self.state = module.params['state']
 
+    def get_interfaces(self):
+        """ Fetch list of interface names present in namespace """
+        rc, ifconfig, err = self._netns_interface()
+        if rc != 0:
+            self.module.fail_json(msg=err)
+        name = r'^\d+:\s(\w+):\s.*'
+        regex = re.compile(name, flags=re.MULTILINE)
+        return regex.findall(ifconfig.decode())
+
     def _netns(self, command):
-        '''Run ip nents command'''
+        """ Run ip nents command """
         return self.module.run_command(['ip', 'netns'] + command)
 
+    def _netns_interface(self, interface=None, operation='ls'):
+        """ List or add or remove interfaces inside namespace """
+        if operation.lower() == 'ls':
+            # list interfaces in the namespace
+            return self.module.run_command(['ip', 'netns', 'exec', self.name, 'ip', 'a', 's'])
```

Comment:
You should not use 'ip a s' here, as it ignores interfaces without L3 addresses. The proper command should be 'ip link list', which shows all interfaces regardless if they are with L3 addresses or without.

Additionally, it's better to use full notation for ip commands (not the shortest abbreviation, as it makes things clearer for reader).